src/docbuilder/rustwide_builder.rs

```diff
@@ -392,9 +392,25 @@ impl RustwideBuilder {
                     }
                 };
 
+                let cargo_metadata = res.cargo_metadata.root();
+
+                // Fetch repository statistics from GitHub
+                let github_updater = GithubUpdater::new(self.config.clone(), self.db.clone())?;
+                let github_repo = if let Some(repo) = &cargo_metadata.repository {
+                    match github_updater.load_repository(&mut conn, repo) {
```

Comment:
this is lazy because `load_repository` is lazy, nice :) (lazy -> uses cached data from the DB if available)